### DISCUSSION TITLE
Vercel Preview Deployment dose not run code in webpack config in next.config.js ?

### DISCUSSION THREAD

**Discussion by anhdd-kuro** [vercel/vercel#8533] [Help]
My next.config.js

```js
/** @type {import('next').NextConfig} */
const nextConfig = {
  reactStrictMode: true,
  env: {
    API_SERVER: process.env.API_SERVER,
  },
  webpack: (config) => {

    console.log(process.env.NODE_ENV, process.env.MOCKED_API);

    if (process.env.NODE_ENV === "production" || process.env.MOCKED_API !== "true") {
      config.module.rules.unshift({
        test: /miragejs\//,
        use: "null-loader",
      });
    }

    return config;
  },
};

module.exports = nextConfig;

```

It's run as expected in local and have log 
But in vercel preview deployment , it dose not run console.log and throw errors 

```
info  - Linting and checking validity of types...
./src/setup/mockedApi/index.ts
Error: Unable to resolve path to module './generated-schema.gql'.  import/no-unresolved 
```
Because this file only be generated in dev mode so in production I'm using null-loader package to ignore import
What am I doing wrong here ?

**Comment by anhdd-kuro** [comment]
I just remove all the import 
```
import { makeServer } from "@/setup/mockedApi";
```
But Vercel still throw same error when run yarn build
So vercel preview deployment linting and checking validity of types for the files that I don't even use ?
Or this is some cache problem ?

**Comment by amyegan** [comment]
Hi @anhdd-kuro! Can you share your repo or [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) so I can take a closer look?
